tests/tests/operation_tests.cpp

```diff
@@ -7607,5 +7607,291 @@ BOOST_AUTO_TEST_CASE( account_auth_tests )
    FC_LOG_AND_RETHROW()
 }
 
+BOOST_AUTO_TEST_CASE( account_update2_validate )
+{
+   try
+   {
+      BOOST_TEST_MESSAGE( "Testing: account_update2_validate" );
+
+      ACTORS( (alice) )
+
+      account_update2_operation op;
+      op.account = "alice";
+      op.posting = authority();
+      op.posting->weight_threshold = 1;
+      op.posting->add_authorities( "abcdefghijklmnopq", 1 );
+
+      try
+      {
+         op.validate();
+
+         signed_transaction tx;
+         tx.set_expiration( db->head_block_time() + STEEM_MAX_TIME_UNTIL_EXPIRATION );
+         tx.operations.push_back( op );
+         sign( tx, alice_private_key );
+         db->push_transaction( tx, 0 );
+
+         BOOST_FAIL( "An exception was not thrown for an invalid account name" );
+      }
+      catch( fc::exception& ) {}
+
+      BOOST_TEST_MESSAGE( " -- Testing failure when json_metadata is not json" );
+      account_update2_operation op1;
+      op1.account = "alice";
+      op1.json_metadata = "not json";
+
+      STEEM_REQUIRE_THROW( op1.validate(), fc::assert_exception );
+
+      BOOST_TEST_MESSAGE( " -- Testing failure when posting_json_metadata is not json" );
+      op1.json_metadata.clear();
+      op1.posting_json_metadata = "not json";
+
+      STEEM_REQUIRE_THROW( op1.validate(), fc::assert_exception );
+      op1.posting_json_metadata.clear();
+
+      op1.validate();
+
+      validate_database();
+   }
+   FC_LOG_AND_RETHROW()
+}
+
+BOOST_AUTO_TEST_CASE( account_update2_authorities )
+{
+   try
```

Comment:
Auth tests are a mixed bag. I wrote most of them during the initial development of Steem while I was still learning about blockchain. Look at a newer operation test, such as `claim_account_authorities` for a better example on how these tests should be structured to be as clear as possible and not test redundant scenarios such as duplicate keys.